functions/src/models/offers/index.ts

```diff
@@ -19,9 +20,11 @@ export interface IOffer extends DocumentData {
   pinUserRef: DocumentReference<DocumentData>;
   requestRef: DocumentReference<DocumentData>;
   cavUserSnapshot: IUser;
+  requestSnapshot: IRequest | null;
   message: string;
   status: OfferStatus;
   createdAt?: Timestamp;
+  seenAt?: Timestamp;
```

Comment:
This should also be allowed to be `null`. firebase doesn't play well with `undefined`

Reply:
+1

Reply:
Gotcha